modmail/exts/utils/plugin_manager.py

```diff
@@ -0,0 +1,276 @@
+# original source:
+# https://github.com/python-discord/bot/blob/a8869b4d60512b173871c886321b261cbc4acca9/bot/exts/utils/extensions.py  # noqa: E501
+# MIT License 2021 Python Discord
+import functools
+import logging
+import typing as t
+from enum import Enum
+from pathlib import Path
+
+from discord import Colour, Embed
+from discord.ext import commands
+from discord.ext.commands import Context, group
+
+from modmail import plugins
+from modmail.bot import ModmailBot
+from modmail.log import ModmailLogger
+from modmail.utils.cogs import ExtMetadata
+from modmail.utils.plugin_manager import PLUGINS, unqualify
+
+log: ModmailLogger = logging.getLogger(__name__)
+
+BASE_PATH = Path(plugins.__file__).parent
+BASE_PATH_LEN = len(plugins.__name__.split("."))
+
+EXT_METADATA = ExtMetadata(production=True, develop=True, plugin_dev=True)
+
+
+class Action(Enum):
+    """Represents an action to perform on an extension."""
+
+    # Need to be partial otherwise they are considered to be function definitions.
+    LOAD = functools.partial(ModmailBot.load_extension)
+    UNLOAD = functools.partial(ModmailBot.unload_extension)
+    RELOAD = functools.partial(ModmailBot.reload_extension)
```

Comment:
Don't define this twice. AS above, this should be a dict. Define it on the ModmailCog class, and then have plugins subclass that.